libgnucash/engine/gnc-numeric.cpp

```diff
@@ -256,6 +259,21 @@ GncNumeric::GncNumeric(const std::string &str, bool autoround) {
         m_den = n.denom();
         return;
     }
+    if (regex_search(str, m, integer_and_fraction))
+    {
+        auto integer{stoll(m[1].str())};
+        if (integer == 0)
+            throw std::invalid_argument ("GncNumeric with integer and fraction, integer cannot be zero");
+        auto frac_num{stoll(m[2].str())};
+        auto frac_denom{stoll(m[3].str())};
+        if (frac_num >= frac_denom)
+            throw std::invalid_argument ("GncNumeric with integer and fraction, fraction should be less than 1");
```

Comment:
Why does that matter?

Reply:
because it seems ridiculous to allow such weirdness like `0 2/3` or `1 8/5`?

Reply:
That's being arbitrary. They're perfectly legitimate numbers and while a human wouldn't write them a lazily-written program might.

Reply:
Perhaps... Then the parser will need to handle such cases like `-0 5/3` being -5/3. The regexes (regices?) will need more upgrades.